src/Microsoft.ML.AutoML/EstimatorExtensions/EstimatorExtensions.cs

```diff
@@ -256,17 +257,45 @@ public IEstimator<ITransformer> CreateInstance(MLContext context, PipelineNode p
             return CreateInstance(context, pipelineNode.InColumns[0], pipelineNode.OutColumns[0]);
         }
 
-        public static SuggestedTransform CreateSuggestedTransform(MLContext context, string inColumn, string outColumn)
+        public static SuggestedTransform CreateSuggestedTransform(MLContext context, string inColumn, string outColumn, bool sort = false)
```

Comment:
Continuation of https://github.com/dotnet/machinelearning/pull/4384#pullrequestreview-307469017

---


> From: @daholste, edited by codemzs 
> out of curiosity, why sort for images? #Resolved
>
> ---
>
>    From: @codemzs
> great question, when you split the dataset and then apply ValueToKeyMapping after the split then it could result in different key to value mapping for the same class. I saw some cases where this was being done in the code base so for image classification I just want to be sure the mappings is consistent else it can degrade the accuracy.
>  
> In reply to: 339251823 [](ancestors = 339251823)
>
> ---
>
>    From: @justinormont
> Why does the mapping being inconsistent affect the model's accuracy?
> 
> You'll run into inconsistent label mappings rather often, specifically the mapping can ends up different each time it runs on a different split of the dataset.
> 
> For instance every model you create, the dataset will be split slightly differently in TrainValidate mode, or within CV. One ValueToKeyMapping run will see 100 classes, another will see 99.
>  
> ---
>
>    From: @codemzs
> I’ll explain offline.

Offline discussion sounds great. Thanks.

Reply:
Why mark as resolved without the discussion?

Reply:
Again, why mark as resolved without the discussion?